### PR TITLE
doc: update api doc

### PR DESCRIPTION
**Pull request by zirain**
fix: https://github.com/envoyproxy/gateway/issues/3414

### CODE REVIEW

File: tools/crd-ref-docs/config.yaml [hunk 1]
```diff
@@ -20,3 +19,10 @@ render:
     - name: LocalPolicyTargetReferenceWithSectionName
       package: sigs.k8s.io/gateway-api/apis/v1alpha2
       link: https://gateway-api.sigs.k8s.io/reference/spec/#gateway.networking.k8s.io/v1alpha2.LocalPolicyTargetReferenceWithSectionName
+    - name: Duration
```
Comment (shawnh2) [inline tools/crd-ref-docs/config.yaml:22]: great! can we do the same things for all the rest types which are ref to gateway-api pkg?

Comment (shawnh2) [reply tools/crd-ref-docs/config.yaml:22]: - https://github.com/envoyproxy/gateway/blob/02ff9eb749ba7eda1a53dfcbd80fb2b6f46b23fe/api/v1alpha1/backendtrafficpolicy_types.go#L46
- https://github.com/envoyproxy/gateway/blob/02ff9eb749ba7eda1a53dfcbd80fb2b6f46b23fe/api/v1alpha1/basic_auth_types.go#L26
- https://github.com/envoyproxy/gateway/blob/02ff9eb749ba7eda1a53dfcbd80fb2b6f46b23fe/api/v1alpha1/ext_auth_types.go#L70
-

File: tools/crd-ref-docs/config.yaml [hunk 2]
```diff
@@ -4,19 +4,34 @@ processor:
...
+      link: https://gateway-api.sigs.k8s.io/reference/spec/#gateway.networking.k8s.io/v1alpha2.LocalPolicyTargetReference
+    - name: Duration
+      package: sigs.k8s.io/gateway-api/apis/v1
+      link: https://gateway-api.sigs.k8s.io/reference/spec/#gateway.networking.k8s.io/v1.Duration
+    - name: PolicyStatus
+      package: sigs.k8s.io/gateway-api/apis/v1alpha2
+      link: https://gateway-api.sigs.k8s.io/reference/spec/#gateway.networking.k8s.io/v1alpha2.PolicyStatus
+    - name: JSON
+      package: k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1
+      link: https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.29/#json-v1-apiextensions-k8s-io
```
Comment (zirain) [inline tools/crd-ref-docs/config.yaml:36]: xref: https://github.com/elastic/crd-ref-docs/pull/82

**Review by shawnh2 — APPROVED** [review]
LGTM! Shall we wait [this PR](https://github.com/elastic/crd-ref-docs/pull/82) to be merged?

**Comment by zirain** [comment]
> LGTM! Shall we wait [this PR](https://github.com/elastic/crd-ref-docs/pull/82) to be merged?

we can do that with a following PR.

**Review by arkodg — APPROVED** [review]
LGTM thanks !

**Pull request merged**